pkg/apis/eventing/v1alpha1/broker_defaults.go

```diff
@@ -0,0 +1,25 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+func (b *Broker) SetDefaults() {
+	b.Spec.SetDefaults()
+}
+
+func (bs *BrokerSpec) SetDefaults() {
+	// None
```

Comment:
If not here then who sets the default channel in case no channel spec is provided. I can see BrokerSpec.ChannelTemplate?

Reply:
If the Broker has an empty `spec.channelTemplate`, then it will create `Channel`'s without any provisioner. The Channel defaulter will be used to default the provisioner.